go/teams/loader.go

```diff
@@ -382,7 +384,7 @@ func (l *TeamLoader) load2Inner(ctx context.Context, arg load2ArgT) (*load2ResT,
 			ret.Secretless = true
 		} else {
 			// Add the secrets unless this is a secretless team.
-			if !ret.Secretless {
+			if !ret.Secretless && !arg.public {
```

Comment:
I think `!ret.Secretless && !ret.Chain.Public` would be a little nicer. Doesn't really matter

Reply:
If they aren't the same, it's an error anyway.

Reply:
yep but if they mismatch it will error here which would be harder to debug